var/spack/repos/builtin/packages/rocm-opencl/package.py

```diff
@@ -163,11 +163,16 @@ def post_install(self):
 
     test_src_dir = "tests/ocltst"
 
-    def test(self):
+    def test_rocm_opencl(self):
+        """Run rocm-opencl tests"""
         test_dir = join_path(self.spec["rocm-opencl"].prefix, self.test_src_dir)
         with working_dir(test_dir, create=True):
             os.environ["LD_LIBRARY_PATH"] += os.pathsep + test_dir
-            args = ["-m", "liboclruntime.so", "-A", "oclruntime.exclude"]
-            self.run_test("ocltst", args)
-            args = ["-m", "liboclperf.so", "-A", "oclperf.exclude"]
-            self.run_test("ocltst", args)
+
+            with test_part(self, "test_rocm_opencl_runtime", purpose="Test runtime"):
+                exe = which("ocltst")
```

Comment:
This only needs to be done once in this case so you can move this above the `test_part()` context manager.

Reply:
Then you don't need to re-instantiate the `Executable` again at line 177.